### ISSUE TITLE
Transition FATES LUH timeseries test from fixed to transient compset

### ISSUE THREAD

**Issue by ekluzek** [ESCOMP/CTSM#2304]
The LUH test added to ctsm5.1.dev160 ERS_D_Ld30.f45_f45_mg37.I2000Clm50FatesCruRsGs.derecho_intel.clm-FatesColdLUH2, is for a fixed compset rather than a transient compset. This should be changed into a transient compset, with do_transient_pft=FALSE. The LUH timeseries by definition is a transient dataset and should be run is a transient case.

In testing this I found that we need to add a transient compset for FATES. I tried using a long compset name, but got a fail in the build: SMS.f45_f45_mg37.HIST_DATM%GSWP3v1_CLM51%FATES_SICE_SOCN_MOSART_SGLC_SWAV.derecho_intel.clm-FatesColdLUH2

**Comment by ekluzek** [comment]
This relates to #1617

**Comment by ekluzek** [comment]
This also makes me wonder if FATES transient cases with LUH can be run with transient lake, or urban?

**Comment by glemieux** [comment]
Thanks for catching this @ekluzek.  Not being very well versed in the compset aliases, is the idea that adding the `HIST_DATM` will tell the build infrastructure to note that this is a historical transient run and thus set the appropriate namelist options and such?

**Comment by ekluzek** [comment]
@glemieux yes, by setting HIST_DATM all the things will be set to run as historical transient. So prescribed-aerosols, and urban, and population density and all those things will be transient.

Actually, I think what we should do is to remove the LUH testmod and set up the HIST FATES compset to have LUH turned on by default. So LUH will be on anytime a HIST compset is used. Possibly we should also not allow LUH to be on for a non HIST compset.

It might be good for us to chat about this at some point?

**Comment by glemieux** [comment]
Thanks @ekluzek.  As you mentioned in the ctsm software meeting today, chatting about this after the 2-stream testing meeting next week sounds like a good plan.

**Comment by ekluzek** [comment]
@glemieux this is what I think this needs to look like.

The transient use-cases turn on flanduse_timeseries files as follows:

``` xml
<clm_demand >flanduse_timeseries</clm_demand>
```

In order to get dynamic lake and urban I think that we leave that as is.
do_transient_pfts will already be turned off for FATES by build-namelist
logic. And transient lake and urban should be appropriately set.

``` xml
<fates_harvest_mode use_fates=".true.">luhdata_mass</fates_harvest_mod>
<use_fates_luh use_fates=".true.">luhdata_mass</use_fates_luh>
```
Note, I'm assuming the default would be luhdata_mass would be the default, but it can be whatever is desired here.

The transient use case will then set urbantv on, and the HIST compset will set DATM forcing to transient as appropriate. ndep and popdens shouldn't matter for FATES. If they do in the future we can add handling for them.

**Comment by ekluzek** [comment]
Some notes about the different ways of handling complex options is here:

https://docs.google.com/presentation/d/1FIaQmgN35v3tQxd9hpxcUQZ2CJXFpfx991rZgXwMKzA

Use-cases are one of those options and talked about there.

**Comment by ekluzek** [comment]
In #2888 I'm bringing in some changes that bring us closer to this for the FatesColdLandUse testmod. This explicitly sets the use-case to the 20thC_transient one to get the needed landuse.timeseries file.

**Comment by ekluzek** [comment]
@glemieux and I talked about this a bit. This is important to get in, because I've already required one of the FATES LUH tests do this. So it will help consistency with those tests and aligning with FATES compsets.

#1617 is also an important usability thing I think we should prioritize for the release.
